cmd/minikube/cmd/config/enable_test.go

```diff
@@ -16,10 +16,20 @@ limitations under the License.
 
 package config
 
-import "testing"
+import (
+	"strings"
+	"testing"
+)
 
 func TestEnableUnknownAddon(t *testing.T) {
 	if err := Set("InvalidAddon", "false"); err == nil {
 		t.Fatalf("Enable did not return error for unknown addon")
 	}
 }
+func TestEnableAddon(t *testing.T) {
+	if err := Set("ingress", "true"); err != nil {
+		if !strings.Contains(err.Error(), "was already enabled") {
```

Comment:
I don't understand this test - why would ingress already be enabled when this test is run? Enabling an enabled addon should not cause an error. We should aim for idempotent API's when possible.

Also, never compare error strings in go: https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully


Also, please don't compare error strings, they are not an API.

Reply:
Is there a way to create a mock machine in a directory? I found .minikube/ test data but they only contain profiles.